frontend/src/MainChat.tsx

```diff
@@ -1,67 +1,61 @@
-            connection.current.onopen = () => {
-                console.log("WebSocket connection established");
-            }
-            connection.current.onclose = () => {
-                console.log("WebSocket connection closed");
-            };
-            connection.current.onmessage = (e) => {
-                setMessagesList(messagesList => [...messagesList, e.data])
-            };
+        if (lastMessage !== null) {
+            setMessageHistory((messageHistory: any) => [...messageHistory, lastMessage]);
         }
-    }, []);
+    }, [lastMessage, setMessageHistory]);
 
-    const keepAlive = useCallback(() => {
-        if (shouldKeepWSAlive.current) {
-            if (connection.current !== undefined &&
-                connection.current !== null &&
-                connection.current.readyState === 1) {
-                connection.current.send("");
-            }
-            setTimeout(() => {
-                keepAlive();
-            }, 20000);
-        }
-    }, []);
+    const handleMessageSubmit = (event: any) => {
+        event.preventDefault();
+        messageToSend();
+    }
 
-    useEffect(() => {
-        if (!connection.current) {
-            shouldKeepWSAlive.current = true;
-            keepAlive();
-        } else {
-            shouldKeepWSAlive.current = false;
-        }
-    }, [keepAlive]);
+    const putDateTime = () => {
+        const date = new Date();
+        return date.toLocaleString();
+    }
+
+    const messageToSend = useCallback(() => sendMessage(message), [sendMessage, message]);
+
+    const connectionStatus = {
+        [ReadyState.CONNECTING]: 'Connecting',
+        [ReadyState.OPEN]: 'Open',
+        [ReadyState.CLOSING]: 'Closing',
+        [ReadyState.CLOSED]: 'Closed',
+        [ReadyState.UNINSTANTIATED]: 'Uninstantiated',
+    }[readyState];
 
-    const messages = messagesList.map((message) => {
-        return (
-            <StyledLi key={nanoid()}>
-                {message}
-            </StyledLi>
-        )
-    });
+    const messages = messageHistory.map((message: { data: string | number | boolean | React.ReactElement<any, string | React.JSXElementConstructor<any>> | React.ReactFragment | React.ReactPortal | null | undefined; }, idx: React.Key | null | undefined) => (
+        <StyledLi key={idx}>
+            {message ? message.data : null}
+        </StyledLi>
+    ));
 
     return <>
-        <StyledH2>Main-Chat</StyledH2>
+        <StyledH2>Main-Chat <span>{connectionStatus}</span></StyledH2>
         <StyledSection>
-        <ul>
-            {messages}
-        </ul>
+            <ul>
+                {messages}
+            </ul>
         </StyledSection>
+
+        <StyledSection2>
+            <form onSubmit={handleMessageSubmit}>
+                <input type="text" disabled={readyState !== ReadyState.OPEN}
+                       onChange={(e) => setMessage(putDateTime() + ": " + e.target.value)}/>
```

Comment:
Ich fände es schöner, wenn der Server den Zeitpunkt ermittelt und speichert. Dann kann man nicht durch einen manipulierten Client Messages in der Vergangenheit oder Zukunft schreiben.

Reply:
jop dann müsste ich aber die protected void handleTextMessage umschreibeln :S überleg ich mir noch :D

Reply:
ok hab ich so gemacht im neuen PR :D https://github.com/chris-yooo/capstone-chrat-reloaded/pull/8/files